### PR TITLE
New custom logger that utilizes pino and the NR enricher

### PR DESCRIPTION
**Pull request by CalvinAllen**

### CODE REVIEW

File: api_server/package.json [hunk 1]
```diff
@@ -48,6 +49,7 @@
 		"node-schedule": "2.1.1",
 		"oauth": "0.9.15",
 		"path": "0.12.7",
+		"pino": "^8.15.0",
```
Comment (bcanzanella) [inline api_server/package.json:52]: nit pick to remove the ^

File: api_server/package.json [hunk 2]
```diff
@@ -3,6 +3,7 @@
 	"version": "14.0.19",
 	"dependencies": {
 		"@microsoft/microsoft-graph-client": "3.0.5",
+		"@newrelic/pino-enricher": "^1.1.1",
```
Comment (bcanzanella) [inline api_server/package.json:6]: nit pick to remove the ^

File: api_server/newrelic.js [hunk 3]
```diff
@@ -7,42 +7,20 @@
...
   license_key: `${process.env.CSSVC_NEWRELIC_LICENSE_KEY}`,
-  /**
-   * This setting controls distributed tracing.
-   * Distributed tracing lets you see the path that a request takes through your
-   * distributed system. Enabling distributed tracing changes the behavior of some
-   * New Relic features, so carefully consult the transition guide before you enable
-   * this feature: https://docs.newrelic.com/docs/transition-guide-distributed-tracing
-   * Default is false.
-   */
-  distributed_tracing: {
```
Comment (bcanzanella) [inline api_server/newrelic.js:26]: is this enabled by default?

File: api_server/newrelic.js [hunk 4]
```diff
@@ -7,42 +7,20 @@
...
   host: 'staging-collector.newrelic.com',
-  /**
-   * Array of application names.
-   */
-  app_name: ['api'],
-  /**
-   * Your New Relic license key.
-   */
+
+  app_name: ['api-apm-test-calvin'],
```
Comment (bcanzanella) [inline api_server/newrelic.js:11]: TODO to change back to 'api'

File: api_server/newrelic.js [hunk 5]
```diff
@@ -7,42 +7,20 @@
...
+
   logging: {
-    /**
-     * Level at which to log. 'trace' is most useful to New Relic when diagnosing
-     * issues with the agent, 'info' and higher will impose the least overhead on
-     * production applications.
-     */
-    level: 'info',
-    filepath: `${process.env.CS_API_LOGS}/newrelic_agent.log`
+    enabled: false,
```
Comment (bcanzanella) [inline api_server/newrelic.js:16]: how come this is false?

**Review by bcanzanella — CHANGES_REQUESTED** [review]

File: api_server/newrelic.js (hunk 3 above)
Comment (CalvinAllen) [reply api_server/newrelic.js:26]: It is with the version we're using

File: api_server/newrelic.js (hunk 5 above)
Comment (CalvinAllen) [reply api_server/newrelic.js:16]: Still playing with this, but that's the newrelic agent log. We can leave that one enabled, assuming it's not part of the overall problem

Comment (CalvinAllen) [reply api_server/newrelic.js]: @bcanzanella Maybe this is a case where we leave this one enabled, and let the infrastructure agent forward it?

**Review by bcanzanella — APPROVED** [review]

**Pull request merged**